PureBasicIDE/tools/LFCheck.pb

```diff
@@ -0,0 +1,427 @@
+﻿; --------------------------------------------------------------------------------------------
+;  Copyright (c) Fantaisie Software. All rights reserved.
+;  Dual licensed under the GPL and Fantaisie Software licenses.
+;  See LICENSE and LICENSE-FANTAISIE in the project root for license information.
+; --------------------------------------------------------------------------------------------
+
+; Localisation File Consistency Checker.
+
+; Usage:
+; LFCheck /u /r="reference file path" /t="target file path" /l="log file path"
+; /u - Check for untranslated text.
+; /r= - Reference catalog path, omit this to reference the IDE's default language table.
+; /t= - Target catalog path.
+; /l= - Log problems to a file.
+
+; Please note:
+; Whilst this tool tells the truth it doesn't necessarily tell the _whole_ truth, particularly with the untranslated terms check active.
+; Terms which do not get translated in langauges other than English or are spelled the same as their English counterpart will be 
+; reported as an untranslated term.
+; Please do not post these as a Documentation or IDE bug without checking first that there really is a problem.
+
+EnableExplicit
+
+;- Enumerations
+EnumerationBinary Origin
+  #Reference
+  #Target
+EndEnumeration
+#Both = #Reference | #Target
+
+EnumerationBinary Options
+  #OptReferenceFile
+  #OptReferenceTable
+  #OptTestUntranslated
+  #OptLogFile
+EndEnumeration
+
+#Key$ = "Key "
+#Group$ = "Group "
+#DefTable$ = "Default Language Table."
+#RefFile$ = "Reference File."
+#TargetFile$ = "Target File."
+
+#SQ$ = "'"
+#MsgTitle$ = "PureBasic Localisation File Consistency Checker"
+#MsgRef$ = "The reference file "
+#MsgTarget$ = "The target file "
+#MsgLog = "The log file "
+#MsgNoTarget$ = "No target file specified."
+#MsgChecking = "Checking: "
+#MsgAgainst = "Against: "
+#MsgNotFound$ = " could not be found."
+#MsgNotOpened$ = " could not be opened."
+#MsgNotCreated$ = " could not be created."
+#MsgNothing$ = "No problems detected."
+
+#ErrEmpty$ = " is empty in the "
+#ErrDup$ = " is duplicated in the "
+#ErrUndef$ = " is not defined in the "
+#ErrMissing$ = " is missing from the "
+#ErrUnTrans$ = " is not translated in the "
+
+;- Structures
+Structure ASTRING
+  State.I
+  RefCount.I
+  TargetCount.I
+  Reference.S
+  Target.S
+EndStructure
+
+;- Variables
+Global.S ReferenceDesc = #DefTable$, ReferenceFile, TargetFile, LogFile
```

Comment:
Type names are not all caps in the IDE code. Also, we consistently use `Variable$` for string variables (you have a bit of a mix here) and leave out the `.i` outside of structures since it is the default type.

The rule is this:
* Strings always as `Variable$`
* Integers always as `Variable`
* Floats or Doubles always as `Variable.d`. Not just at the declaration but also when used to make it more visible that it is not an integer variable. Floating point variables are rare, though.